MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,178 +1,158 @@
+        noButton.isEnabled = true
+        yesButton.isEnabled = true
+    }
+    
+    private func blockButton() {
+        noButton.isEnabled = false
+        yesButton.isEnabled = false
     }
     
-    //состояние показа результата ответа
+    private func show(quiz step: QuizStepViewModel) {
+        imageView.image = step.image
+        imageView.layer.cornerRadius = 20
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+    }
+    
+    
     private func showAnswerResult(isCorrect: Bool) {
-        //счетчик правильных ответов
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.cornerRadius = 20
         if isCorrect {
+            imageView.layer.borderColor = UIColor.ypGreen.cgColor
             correctAnswers += 1
+        } else {
+            imageView.layer.borderColor = UIColor.ypRed.cgColor
         }
-        //Рамки в зеленый
-        imageView.layer.masksToBounds = true
-        imageView.layer.borderWidth = 8
-        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
-        
-        // Запуск следующего вопроса через 1 секунду
-        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
-            self.imageView.layer.borderColor = UIColor.clear.cgColor
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [ weak self ] in
+            guard let self = self else { return } // optional weak link is commonly deployed
             self.showNextQuestionOrResults()
+            self.imageView.layer.borderWidth = 0
+            self.unlockButton()
         }
     }
     
-    //показ следующего вопроса или результата
-    private func showNextQuestionOrResults(){
-        if currentQuestionIndex == questions.count - 1 {
-            // показать результат
-            let text = "Ваш результат: \(correctAnswers) из 10"
-            let viewModel = QuizResultsViewModel(title: "Этот раунд окончен", text: text, buttonText: "Сыграть еще раз")
-            show(quiz: viewModel)
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(image: UIImage(named: model.image) ?? UIImage(),
+                                 question: model.text,
+                                 questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)")
+    }
+    
+    // MARK: showNextQuestionOrResults
+    private func showNextQuestionOrResults() {
+        imageView.layer.borderWidth = 0
+        if currentQuestionIndex == questionsAmount - 1 {
+            imageView.layer.borderWidth = 8
+            statisticService?.store(correct: correctAnswers, total: questionsAmount)
+            guard let gamesCount = statisticService?.gamesCount else { return }
+            guard let bestGame = statisticService?.bestGame else { return }
+            guard let totalAccuracy = statisticService?.totalAccuracy else { return }
```

Comment:
можно разыменовать statisticService, чтобы не создавать промежуточные переменные под его свойства
 
```suggestion
            guard let statisticService else { return } 
            statisticService.store(correct: correctAnswers, total: questionsAmount)
            ...
```